File: src/converter/annotation.ts
import {ConverterContext} from "./context.js";
import {Node} from "typescript"

export interface AnnotationContext extends ConverterContext {
    isAnonymousDeclaration: boolean
}

export type Annotation<TNode extends Node = Node> =
    (node: TNode, context: AnnotationContext) => string | null

File: gradle-plugin/karakum/src/jsMain/kotlin/io/github/sgrishchenko/karakum/extension/Annotation.kt
package io.github.sgrishchenko.karakum.extension

import seskar.js.JsNative
import typescript.Node

external interface AnnotationContext : ConverterContext {
    val isAnonymousDeclaration: Boolean
}

external interface Annotation<in TNode : Node> {
    @JsNative
    operator fun invoke(node: TNode, context: AnnotationContext): String
}
